core/variant/binder_common.h

```diff
@@ -224,6 +224,14 @@ struct PtrToArg<char32_t> {
 template <typename T>
 struct VariantObjectClassChecker {
 	static _FORCE_INLINE_ bool check(const Variant &p_variant) {
+#ifdef DEBUG_ENABLED
+		bool freed;
+		p_variant.get_validated_object_with_check(freed);
+		if (freed) {
+			return false;
+		}
+#endif
+
 		using TStripped = std::remove_pointer_t<T>;
 		if constexpr (std::is_base_of<Object, TStripped>::value) {
 			Object *obj = p_variant;
```

Comment:
change the line just below this to `return !obj || Object::cast_to<TStripped>(p_variant);`